lib/logstash/filters/csv.rb

```diff
@@ -151,10 +151,14 @@ def filter(event)
   private
 
   # construct the correct Event field reference for given field_name, taking into account @target
-  # @param field_name [String] the bare field name without brakets
+  # @param field_name [String] the field name.
   # @return [String] fully qualified Event field reference also taking into account @target prefix
   def field_ref(field_name)
-    "#{@target}[#{field_name}]"
+    if field_name.start_with?("[")
```

Comment:
I added 3 tests:

* Nested columns but no target (https://github.com/logstash-plugins/logstash-filter-csv/pull/52/files#diff-9f0c7e51c98d7c9b263691b6cdf82d58R214)
* Nested target but no columns (https://github.com/logstash-plugins/logstash-filter-csv/pull/52/files#diff-9f0c7e51c98d7c9b263691b6cdf82d58R264)
* Nested target and nested columns (https://github.com/logstash-plugins/logstash-filter-csv/pull/52/files#diff-9f0c7e51c98d7c9b263691b6cdf82d58R277)